### ISSUE TITLE
SE3.Eul() method returns same for 180 degree rotation about X and Z axes

### ISSUE THREAD

**Issue by jmount1992** [petercorke/spatialmath-python#45]
The [SE3.Eul method](https://github.com/petercorke/spatialmath-python/blob/feeaa3639699359e3af0c2b29949edde096af8f2/spatialmath/pose3d.py#L414) returns the same matrix for a 180 degree rotation about the X and Z axes.

Running
```
import spatialmath as sm
print(sm.SE3.Eul([0, 0, 180], unit='deg')) # this creates same rotation matrix as line below
print(sm.SE3.Eul([180, 0, 0], unit='deg'))
```

Haven't had a chance to look further as to why. Will hopefully find the root of the problem and do a PR.

**Comment by petercorke** [comment]
This is correct behaviour.  Because the middle rotation is zero, it is a rotation of 180deg about z-axis in both cases.
